### PR TITLE
feat: Introduce Azure Event Grid as a new CloudEvent destination

### PR DESCRIPTION
**Pull request by SpiritZhou**
Update doc about new AzureEventGrid destination

### Checklist

- [X] Commits are signed with Developer Certificate of Origin (DCO)

Relates to https://github.com/kedacore/keda/issues/3587 
Relates to https://github.com/kedacore/keda/pull/5606

### CODE REVIEW

File: content/docs/2.14/operate/events.md [hunk 1]
```diff
@@ -49,6 +49,9 @@ spec:
   destination:
     http:
       uri: http://foo.bar
+    azureEventGrid:
```
[the remark this answers is not in the document]
Comment (SpiritZhou) [reply content/docs/2.14/operate/events.md:52]: namespace isn't supported rightnow. To support namespace should use another lib. Topic and namespace are using two different libs.

[the remark this answers is not in the document]
Comment (tomkerkhove) [reply content/docs/2.14/operate/events.md]: Exactly my point :) So let's clarify that in the name and add topic in there

File: content/docs/2.14/operate/events.md [hunk 2]
```diff
@@ -75,13 +78,22 @@ There will be multiple types of destination to emit KEDA events to.
...
 ```yaml
   destination:
     http:
       uri: http://foo.bar  #An http endpoint that can receive cloudevent
 ```
+#### Azure Event Grid
+```yaml
+  destination:
+    azureEventGrid:
+      endPoint: Endpoint:foo.bar #endpoint from AzureEventGrid Topic
```
Comment (tomkerkhove) [inline content/docs/2.14/operate/events.md:93]: Let's fix the casing, can we also use a proper event grid topic example?
```suggestion
      endpoint: Endpoint:foo.bar #endpoint from AzureEventGrid Topic
```

File: content/docs/2.14/operate/events.md [hunk 3]
```diff
@@ -75,13 +78,22 @@ There will be multiple types of destination to emit KEDA events to.
...
+      key: *** #accesskey to  AzureEventGrid Topic
```
Comment (tomkerkhove) [inline content/docs/2.14/operate/events.md:94]: Let's update to use trigger authentication

File: content/docs/2.14/operate/events.md [hunk 4]
```diff
@@ -49,6 +49,9 @@ spec:
   destination:
     http:
       uri: http://foo.bar
+    azureEventGridTopic:
+      endpoint: https://my-topic.eastus-1.eventgrid.azure.net/api/events
+      key: ***
```
Comment (tomkerkhove) [inline content/docs/2.14/operate/events.md:54]: I believe this needs to be removed

File: content/docs/2.14/operate/events.md [hunk 5]
```diff
@@ -75,13 +78,40 @@ There will be multiple types of destination to emit KEDA events to.
...
+```yaml
+  destination:
+    azureEventGrid:
+      endpoint: Endpoint:foo.bar #endpoint from AzureEventGrid Topic
+```
+```TriggerAuthentication``` is needed to provide accesskey or identity for Azure Evnet Grid authentication.
+
+**Connection String Authentication:**
+
+- `key` - accesskey string for the Azure Event Grid auth.
```
Comment (tomkerkhove) [inline content/docs/2.14/operate/events.md:99]:
```suggestion
- `key` - Access key string for the Azure Event Grid auth.
```

Is this conenctio nstring or?

File: content/docs/2.14/operate/events.md [hunk 6]
```diff
@@ -75,13 +78,40 @@ There will be multiple types of destination to emit KEDA events to.
...
     http:
       uri: http://foo.bar  #An http endpoint that can receive cloudevent
 ```
+#### Azure Event Grid
+```yaml
+  destination:
+    azureEventGrid:
+      endpoint: Endpoint:foo.bar #endpoint from AzureEventGrid Topic
+```
+```TriggerAuthentication``` is needed to provide accesskey or identity for Azure Evnet Grid authentication.
```
Comment (tomkerkhove) [inline content/docs/2.14/operate/events.md:95]: How do I link these two?

**Review by tomkerkhove — CHANGES_REQUESTED** [review]

File: content/docs/2.14/operate/events.md (hunk 5 above)
Comment (SpiritZhou) [reply content/docs/2.14/operate/events.md:98]: It's an access key that requested with endpoint. It's not a connection string.

File: content/docs/2.14/operate/events.md (hunk 6 above)
Comment (SpiritZhou) [reply content/docs/2.14/operate/events.md:94]: KEDA will try to find if TA has a `key` value.

Comment (tomkerkhove) [reply content/docs/2.14/operate/events.md:101]: That is not how it should work though, the end-user should specify the trigger authentication resource to use or there is a security risk

File: content/docs/2.14/operate/events.md [hunk 7]
```diff
@@ -46,9 +46,15 @@ metadata:
   name: {cloud-event-name}
 spec:
   clusterName: {cluster-name} #Optional. Will be used in the source/subject to specify where the event comes from. The default value is 'kubernetes-default' and it can also be set during the installation of KEDA with --k8sClusterName. This one will overwrite others if set.
+  authenticationRef: 
+    name: {trigger-authentication-name} #Optional. Used to reference a `TriggerAuthentication` for authentication. 
+    kind: ClusterTriggerAuthentication #Optional. Used to choose the authenticatio scopes. https://keda.sh/docs/2.13/concepts/authentication/#authentication-scopes-namespace-vs-cluster
```
Comment (tomkerkhove) [inline content/docs/2.14/operate/events.md:51]: Let's use namespace-scoped TA by default
```suggestion
    name: {trigger-authentication-name} #Optional. Used to reference a `TriggerAuthentication` for authentication. 
    kind: TriggerAuthentication # Optional. Used to choose the authentication scopes. https://keda.sh/docs/latest/concepts/authentication/#authentication-scopes-namespace-vs-cluster
```

Is `name` really optional? I presume it is only when `azureEventGridTopic` is not used?

File: content/docs/2.14/operate/events.md [hunk 8]
```diff
@@ -88,6 +95,31 @@ Here is an overview of the supported destinations:
       uri: http://foo.bar  #An http endpoint that can receive cloudevent
 ```
 
+#### Azure Event Grid
+```yaml
+  destination:
+    azureEventGrid:
+      endpoint: Endpoint:foo.bar #endpoint from AzureEventGrid Topic
+```
+```TriggerAuthentication``` is needed to provide accesskey or identity for Azure Event Grid authentication. ```authenticationRef``` should be configured when sinking to Azure Event Grid
```
Comment (tomkerkhove) [inline content/docs/2.14/operate/events.md:104]:
```suggestion

Authentication information must be provided by using `authenticationRef` which allows you to provide the access key or managed identity for Azure Event Grid authentication by providing a `TriggerAuthentication`.

Here is an overview of the supported authentication types:
```

File: content/docs/2.14/operate/events.md [hunk 9]
```diff
@@ -88,6 +95,31 @@ Here is an overview of the supported destinations:
       uri: http://foo.bar  #An http endpoint that can receive cloudevent
 ```
 
+#### Azure Event Grid
+```yaml
```
Comment (tomkerkhove) [inline content/docs/2.14/operate/events.md:99]: Nit
```suggestion

```yaml
```

File: content/docs/2.14/operate/events.md [hunk 10]
```diff
@@ -88,6 +95,31 @@ Here is an overview of the supported destinations:
...
+  destination:
+    azureEventGrid:
+      endpoint: Endpoint:foo.bar #endpoint from AzureEventGrid Topic
+```
+```TriggerAuthentication``` is needed to provide accesskey or identity for Azure Event Grid authentication. ```authenticationRef``` should be configured when sinking to Azure Event Grid
+
+**Connection String Authentication:**
```
Comment (tomkerkhove) [inline content/docs/2.14/operate/events.md:106]:
```suggestion
##### Connection String Authentication
```

File: content/docs/2.14/operate/events.md [hunk 11]
```diff
@@ -88,6 +95,31 @@ Here is an overview of the supported destinations:
...
+
+- `accessKey` - Access key string for the Azure Event Grid connection auth.
+
+**Pod identity based authentication:**
+[Azure AD Workload Identity](https://azure.github.io/azure-workload-identity/docs/) providers can be used.
+
+```yaml
```
Comment (tomkerkhove) [inline content/docs/2.14/operate/events.md:113]:
```suggestion
##### Pod identity based authentication
[Azure AD Workload Identity](https://azure.github.io/azure-workload-identity/docs/) providers can be used.

```yaml
```

File: content/docs/2.14/operate/events.md [hunk 12]
```diff
@@ -88,6 +95,31 @@ Here is an overview of the supported destinations:
...
+apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: nameOfTriggerAuth
+  namespace: default
+spec:
+  podIdentity:
+    provider: azure-workload
```
Comment (tomkerkhove) [inline content/docs/2.14/operate/events.md:121]: Let's close the code sample

File: content/docs/2.14/operate/events.md [hunk 13]
```diff
@@ -88,6 +95,31 @@ Here is an overview of the supported destinations:
...
+```yaml
+  destination:
+    azureEventGrid:
+      endpoint: Endpoint:foo.bar #endpoint from AzureEventGrid Topic
+```
+```TriggerAuthentication``` is needed to provide accesskey or identity for Azure Event Grid authentication. ```authenticationRef``` should be configured when sinking to Azure Event Grid
+
+**Connection String Authentication:**
+
+- `accessKey` - Access key string for the Azure Event Grid connection auth.
```
Comment (tomkerkhove) [inline content/docs/2.14/operate/events.md:108]: To verify - `accessKey` is the expected key from TA, right?

File: content/docs/2.14/operate/events.md (hunk 7 above)
Comment (SpiritZhou) [reply content/docs/2.14/operate/events.md:51]: It's optional. I reused the ta logic of ScaledObejct. So all properties requirement are the same as ScaledObejct.

File: content/docs/2.14/operate/events.md (hunk 13 above)
Comment (SpiritZhou) [reply content/docs/2.14/operate/events.md:112]: Yes. TA needs the `accessKey`. And KEDA will use it to auth with connection string.

**Review by tomkerkhove — APPROVED** [review]
LGTM, waiting for feature to be merged

**Pull request merged**